notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/NotificationHub.java

```diff
@@ -519,4 +523,28 @@ public static InstallationTemplate getTemplate(String templateName) {
     public InstallationTemplate getInstanceTemplate(String templateName) {
         return mTemplateVisitor.getTemplate(templateName);
     }
+
+    /**
+     *
+     * @param userId User Id
+     */
+    public static void setUserId(String userId) {
+        getInstance().setInstanceUserId(userId);
+    }
+
+    public void setInstanceUserId(String userId) {
+        mUserIdVisitor.setUserId(userId);

```

Comment:
Once the visitor informs us of whether or not there was a change, we should read that value here. If there was a change, we should call `beginInstallationUpdate`. You can look at `removeInstanceTags` for an example.